libs/ngxtension/click-outside/src/click-outside.ts

```diff
@@ -0,0 +1,62 @@
+import {
+	Directive,
+	ElementRef,
+	EventEmitter,
+	inject,
+	Injectable,
+	NgZone,
+	Output,
+} from '@angular/core';
+
+import type { OnInit } from '@angular/core';
+import { injectDestroy } from 'ngxtension/inject-destroy';
+import { fromEvent, Subject, takeUntil } from 'rxjs';
+
+/*
+ * This service is used to detect clicks in the document.
+ * It is used by the clickOutside directive.
+ */
+@Injectable({ providedIn: 'root' })
+export class DocumentClickService {
+	click$ = new Subject<Event>();
+
+	constructor(ngZone: NgZone) {
```

Comment:
nit: Can you please use the inject function here instead? This is purely to be consistent with the rest of the files, including this one, where below we can see inject being used.

Can be skipped if it's too much trouble.

Reply:
Thanks. I will update based on @nartc suggestion.